cli/azd/cmd/middleware/command_hooks.go

```diff
@@ -0,0 +1,86 @@
+package middleware
+
+import (
+	"context"
+	"log"
+
+	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/internal"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/ext"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+)
+
+func UseCommandHooks() actions.MiddlewareFn {
+	return func(ctx context.Context, options *actions.ActionOptions, next actions.NextFn) (*actions.ActionResult, error) {
+		azdContext, err := azdcontext.NewAzdContext()
```

Comment:
shouldn't the context be a dependency for using a hook?

Reply:
Are you referring to the go context or the azd context? There wasn't any need for middleware function to require a context so it wasn't required right now.  But this `UseCommandHooks()` method is the entrypoint for this middleware.  If the developer requires additional inputs for the middleware registration then can include anything they want here including context, configuration options, etc.

Reply:
Or are you referring to why I need to manually create a `NewAzdContext` here?  If that is the question it is because I didn't want all middlewares to have a laundry list of dependencies that "might" be used but rather have the middleware author get access to the components they need.  

This might be a good case for another middleware that registers our most used dependencies in the context that is passed into the executing middleware.